### PR TITLE
Task 3. React Routers

### PR DESCRIPTION
**Pull request by Skilsar**
# Overview
Assignments Task 3. React routers

# Changes
- Added react router navigation in a project;
- Added 4 demo pages;
- Added demo navigation of pages.

# How to see
- GitHub Pages: https://skilsar.github.io/iTechArt-StudentLab/

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
-import { GlobalStyle } from './styled';
+import { GlobalStyles } from './styled';
+
+import './App.css';
 
 const App = (): JSX.Element => (
   <ConfigProvider theme={themeConfig}>
-    <GlobalStyle />
-    <DefaultLayout />
+    <GlobalStyles />
```
Comment (dapodev) [inline src/App.tsx:14]: Do we still need GlobalStyles?

File: src/components/Navigate/Navigate.tsx [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
 import { FC } from 'react';
+import { Link } from 'react-router-dom';
 
 import Logo from 'assets/images/logo.png';
+import { ROUTE } from 'router/routes';
 
 import { StyledHeader, StyledLogo, StyledMenuItem } from './styled';
 
-const MENU_ITEMS = ['All Users', 'All Posts', 'About project'];
+const MENU_ITEMS = [
```
Comment (dapodev) [inline src/components/Navigate/Navigate.tsx:10]: extract constant to another file

File: src/components/Navigate/Navigate.tsx [hunk 3]
```diff
@@ -1,11 +1,18 @@
...
+  { name: 'Posts', link: ROUTE.posts },
+  { name: 'Profile', link: ROUTE.profile },
+  { name: 'SignUp', link: ROUTE.signUp },
+  { name: 'SignIn', link: ROUTE.signIn },
+];
 const DEFAULT_SELECTED_MENU_ITEM = '0';
```
Comment (dapodev) [inline src/components/Navigate/Navigate.tsx:16]: And that one too

File: src/components/Title/Title.tsx [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { FC } from 'react';
+
+import { TitleProps } from 'types/titles/title';
```
Comment (dapodev) [inline src/components/Title/Title.tsx:3]: Why this type is in global types folder? Props types are declared within component folder

File: src/pages/Error/Error.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import { FC } from 'react';
+
+import NotFound from 'assets/images/not-found.svg';
```
Comment (dapodev) [inline src/pages/Error/Error.tsx:3]: Could you please not import sag directly? you can declare index file inside assets/images folder and use that notation: 
`export { ReactComponent as NotFoundImage } from './not-found.svg';`

File: src/pages/Error/Error.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+} from './styled';
+
+const Error: FC = () => (
+  <StyledContent>
+    <StyledWrapper>
+      <StyledTitle>404</StyledTitle>
+      <StyledDescription>
+        The page you are looking for doesn&apos;t exist.
+      </StyledDescription>
+      <StyledButton href='/'>Back to home</StyledButton>
```
Comment (dapodev) [inline src/pages/Error/Error.tsx:22]: Magic string

File: src/pages/Posts/Posts.tsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
+import { FC } from 'react';
+
+import Title from 'components/Title';
+import PostsList from 'components/PostsList';
+
+import { StyledContent } from './styled';
+
+const POSTS = [
```
Comment (dapodev) [inline src/pages/Posts/Posts.tsx:8]: to constants file

File: src/pages/SignIn/styled.ts [hunk 8]
```diff
@@ -0,0 +1,16 @@
+import { Layout } from 'antd';
+import styled from 'styled-components';
+
+import { COLORS } from 'config/colors';
+import { DEFAULT_STYLE } from 'config/defaultStyle';
+
+const StyledContent = styled(Layout.Content)({
+  background: COLORS.background,
```
Comment (dapodev) [inline src/pages/SignIn/styled.ts:8]: Styles order!!!!!! Please, Fix all over the PR

File: src/router/router.tsx [hunk 9]
```diff
@@ -0,0 +1,28 @@
+import { Navigate, Route, Routes } from 'react-router-dom';
```
Comment (dapodev) [inline src/router/router.tsx:1]: Filename has lowercase

File: src/router/routes.ts [hunk 10]
```diff
@@ -0,0 +1,9 @@
+export enum ROUTE {
```
Comment (dapodev) [inline src/router/routes.ts:1]: Enum constants should not be UPPERCASED

File: src/router/router.tsx (hunk 9 above)
Comment (dapodev) [reply src/router/router.tsx:1]: BTW this component is more like RouteList, but not a router. Could you please rename it?

File: src/router/router.tsx [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+import ErrorPage from 'pages/Error';
+import DefaultLayout from 'components/DefaultLayout';
+
+import { ROUTE } from './routes';
+
+const Router = () => {
+  return (
+    <Routes>
+      <Route element={<DefaultLayout />}>
+        <Route path={ROUTE.main} element={<Navigate to={ROUTE.posts} />} />
```
Comment (dapodev) [inline src/router/router.tsx:16]:
```suggestion
        <Route index element={<Navigate to={ROUTE.posts} />} />
```

**Review by dapodev — CHANGES_REQUESTED** [review]
Not found page logics is not working, please, visit [link](https://skilsar.github.io/iTechArt-StudentLab/not-found)

File: src/App.tsx (hunk 1 above)
Comment (Skilsar) [reply src/App.tsx:14]: Sori, I forgot)

**Comment by Skilsar** [comment]
@dapodev, I fixed GitHub Pages

**Review by dapodev — CHANGES_REQUESTED** [review]
Not found page still doesn't work 😢

File: src/App.tsx [hunk 12]
```diff
@@ -1,14 +1,18 @@
...
+import { BrowserRouter } from 'react-router-dom';
 
-import DefaultLayout from 'components/DefaultLayout';
+import Router from 'components/RouteList/RouteList';
 
 import { themeConfig } from './config/themeConfig';
-import { GlobalStyle } from './styled';
+import './App.css';
+
+const BASE_REPO_NAME = '/iTechArt-StudentLab';
```
Comment (dapodev) [inline src/App.tsx:9]: That should be provided using `.env`

File: src/pages/Error/styled.ts [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+  display: 'flex',
+  flexDirection: 'column',
+  justifyContent: 'center',
+  alignItems: 'center',
+  backgroundColor: COLORS.background,
+  textAlign: 'center',
+});
+
+const StyledTitle = styled('h1')({
+  margin: '0 30px 0',
```
Comment (dapodev) [inline src/pages/Error/styled.ts:22]:
```suggestion
  margin: '0 30px',
```

**Review by dapodev — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 14]
```diff
@@ -4,9 +4,10 @@ import { BrowserRouter } from 'react-router-dom';
 import Router from 'components/RouteList/RouteList';
 
 import { themeConfig } from './config/themeConfig';
+
 import './App.css';
 
-const BASE_REPO_NAME = '/iTechArt-StudentLab';
+const BASE_REPO_NAME = process.env.REACT_APP_BASE_REPO_NAME;
```
Comment (dapodev) [inline src/App.tsx:10]: No validation for `env` variables? That's a bad practice

**Review by dapodev — CHANGES_REQUESTED** [review]

**Pull request merged**